PolyLook: Add tokens for themes and buttons

mteresa-jimenez opened the pull request:
# ✍️ Description
<!-- 
    Please, include a summary of the changes you made here and the related issue.
    Some questions to help you give more context:
     - What is the current behavior?
     - Why we need this change/addition?
     - What is the new behavior (if this a feature change)?
     - What should we expect after this change/addition? 
     - How did you test it?
     - Does this introduce a breaking change?
     - Are there any dependencies that are required for this change?
 -->

- [x] **Create "coop" theme in PolyLook** and added in Membership feature with a button as an example. 
        `--theme-primary-color` is the background color, since that variable is already used in the Screen component.
        `--theme-secondary-color` is the text color
        `--theme-tertiary-color` is `--color-red-300` for the three themes that we already have.
     
- [ ] **Add buttons' tokens and implement them in PolyButton**

- [ ] **Fix button types in PolyButton**: "primary" and "secondary" buttons, instead of "filled" and "outline".
       There is something that I am not sure how to address here, it's what I've called the `primary-medium` button.


<!-- Link here the related JIRA issue in case of a bug / feature -->
### 🏗️ Fixes (#jira-issue-number or link)


<!-- Optional -->
## ℹ️ Other information
<!-- 
    Any other information that is important to this PR. 
    e.g. if regards a frontend UI change, add some screenshots 📸
    of how the app looked before and how it looks after the change.
-->


<!-- Please, delete comments when you are done -->

♥️ Thank you!

victorjanin commented:
Great to see this taking form! 💯 

Here are some questions/comments

1. The coop theme should live in the membership feature no? 
2. theme-primary-color naming could be improved to make it more relevant to the background-color (if that is what it is meant for). Otherwise a primary color would be used for the fill of the button, and the fill of an icon for example.
3. The primary button only has (and should have) one color throughout the feature and that would be the primary-color. If there are 2 primary buttons with different fill colors then we need to revisit the design. So I don't think there is the need for  primary-medium button (the name already indicates that there is an issue somewhere, there shouldn't be anything between primary and secondary) 

We could also sync to discuss it if you want, just ping me whenever.

g3o314 commented on feature-utils/poly-look/src/css/themes.css:
```diff
@@ -21,15 +21,37 @@
+  --pl-button-color-text-secondary: var(--theme-secondary-color);
+}
+
+.poly-theme-coop {
```
I still have to finish looking through/thinking about this, but in the meantime, the "coop" theme should be moved from poly-look to the feature since it's feature specific. If at some point we decide to re-use it, we can just move it back.

victorjanin replied:
Tagging along here, I agree, and also, this way you only need to change the values of 
- --theme-primary-color
- --theme-secondary-color
- --theme-tertiary-color
since the button is already taking in those variables

That would be the power of using alias tokens  and global tokens.

mteresa-jimenez replied:
As I commented previously, I still don't understand very well if we're having two features or just one with this new theme. However, independent of that, the `Screen` component that uses `--theme-primary-color` as background-color and the `PolyButton` are in poly-look.

g3o314 commented on feature-utils/poly-look/src/css/themes.css:
```diff
@@ -21,15 +21,37 @@
   --theme-secondary-color: var(--color-indigo-950);
+  --theme-tertiary-color: var(--color-red-300);
   --theme-small-print-color: var(--color-gray-500);
-  --theme-info-button-color: var(--color-gray-300);
```
this one is used by the icon button and it always has the same color, I think it's safe to move it under `poly-theme`

g3o314 commented on feature-utils/poly-look/src/react-components/buttons/polyButton.jsx:
```diff
@@ -8,15 +8,16 @@ import "./polyButton.css";
  */
 
 const types = {
-  filled: "filled",
```
I would revert this change. Ultimately the type is not about the colors but whether if you want a filled or outlined button.
For the colors, there are multiple ways to override the defaults. If it doesn't match the theme configuration, we can always pass a css class to the component or maybe use locally scooped css variables to override the colors.

mteresa-jimenez commented:
> Great to see this taking form! 💯
> 
> Here are some questions/comments
> 
> 1. The coop theme should live in the membership feature no?
> 2. theme-primary-color naming could be improved to make it more relevant to the background-color (if that is what it is meant for). Otherwise a primary color would be used for the fill of the button, and the fill of an icon for example.
> 3. The primary button only has (and should have) one color throughout the feature and that would be the primary-color. If there are 2 primary buttons with different fill colors then we need to revisit the design. So I don't think there is the need for  primary-medium button (the name already indicates that there is an issue somewhere, there shouldn't be anything between primary and secondary)
> 
> We could also sync to discuss it if you want, just ping me whenever.

1. Yes, I see that @g3o314 thinks the same. I thought of doing it in `poly-look` so it could be reused. The thing is, not related to this topic, it's still unclear to me if we're going to have two features which would have this theme (`Membership` and `GDPR Helper`). If we're having just one, I'll move it to the feature.

2. Regarding which one is (or should be) the "primary-color", in the case of the the "light" and "dark" themes is currently assigned be the background-color, just because we thought it makes sense that the "primary-color" in the light theme is white, and indigo for the dark-theme. (Wouldn't that be the logic? 🤔).
Knowing that, we have a `Screen` component and we assigned the variable `--theme-primary-color` to the background. For that reason, in order to make this work and not changing how that component works, I assigned that "primary-color" to the light red.
However, we could also create an extra variable for the whole screen background-color and implement it in the `Screen` component. In that case, background-color doesn't need to always match the "primary-color". 

3. In the importers, the `primary-button` in the "light-theme" is different in different parts of the feature. In the "import" view, they are red, and in the "How to read this" screens, the OK button is indigo. As Geo has also commented, we were just changing the color passing a className, that is really easy to implement (and actually I'm deleting that "primary-medium" since it doesn't make sense and change it back as how it was). But let us know if the design is revisit at some point so we can change that.

victorjanin commented:
> > Great to see this taking form! 💯
> > Here are some questions/comments
> > 
> > 1. The coop theme should live in the membership feature no?
> > 2. theme-primary-color naming could be improved to make it more relevant to the background-color (if that is what it is meant for). Otherwise a primary color would be used for the fill of the button, and the fill of an icon for example.
> > 3. The primary button only has (and should have) one color throughout the feature and that would be the primary-color. If there are 2 primary buttons with different fill colors then we need to revisit the design. So I don't think there is the need for  primary-medium button (the name already indicates that there is an issue somewhere, there shouldn't be anything between primary and secondary)
> > 
> > We could also sync to discuss it if you want, just ping me whenever.
> 
> 1. Yes, I see that @g3o314 thinks the same. I thought of doing it in `poly-look` so it could be reused. The thing is, not related to this topic, it's still unclear to me if we're going to have two features which would have this theme (`Membership` and `GDPR Helper`). If we're having just one, I'll move it to the feature.
> 2. Regarding which one is (or should be) the "primary-color", in the case of the the "light" and "dark" themes is currently assigned be the background-color, just because we thought it makes sense that the "primary-color" in the light theme is white, and indigo for the dark-theme. (Wouldn't that be the logic? 🤔).
>    Knowing that, we have a `Screen` component and we assigned the variable `--theme-primary-color` to the background. For that reason, in order to make this work and not changing how that component works, I assigned that "primary-color" to the light red.
>    However, we could also create an extra variable for the whole screen background-color and implement it in the `Screen` component. In that case, background-color doesn't need to always match the "primary-color".
> 3. In the importers, the `primary-button` in the "light-theme" is different in different parts of the feature. In the "import" view, they are red, and in the "How to read this" screens, the OK button is indigo. As Geo has also commented, we were just changing the color passing a className, that is really easy to implement (and actually I'm deleting that "primary-medium" since it doesn't make sense and change it back as how it was). But let us know if the design is revisit at some point so we can change that.

1. Yes, it's a bit unclear if we have one or 2 features, please bring up this question on Wednesday because I'm not sure what the final decision was, and it's good that everyone hears it. Nonetheless, the token concepts worked on here should work in both cases. 
2. I would be in favor of creating a token for the background color as "primary-color" is often reserved to the main action color. In any case I would rename it to make it clear that it's signifying the background color. Because a primary-color could be applied in many places, but a screen background color is more specific. Let me know if I can assist on naming.
3. Ah ok I got it, I thought in this membership feature there were 2 different colors. When working with on-dark or on-light themes then that could work, we just need another set of tokens for it. I wanted to avoid doing so for the sake of simplifying this token exercise. On the other hand, there is no reason why we should have a red button in the importer. I will challenge that to see where it came from

g3o314 commented:
> > 1. Yes, I see that @g3o314 thinks the same. I thought of doing it in `poly-look` so it could be reused. The thing is, not related to this topic, it's still unclear to me if we're going to have two features which would have this theme (`Membership` and `GDPR Helper`). If we're having just one, I'll move it to the feature.
> > 2. Regarding which one is (or should be) the "primary-color", in the case of the the "light" and "dark" themes is currently assigned be the background-color, just because we thought it makes sense that the "primary-color" in the light theme is white, and indigo for the dark-theme. (Wouldn't that be the logic? 🤔).
> >    Knowing that, we have a `Screen` component and we assigned the variable `--theme-primary-color` to the background. For that reason, in order to make this work and not changing how that component works, I assigned that "primary-color" to the light red.
> >    However, we could also create an extra variable for the whole screen background-color and implement it in the `Screen` component. In that case, background-color doesn't need to always match the "primary-color".
> > 3. In the importers, the `primary-button` in the "light-theme" is different in different parts of the feature. In the "import" view, they are red, and in the "How to read this" screens, the OK button is indigo. As Geo has also commented, we were just changing the color passing a className, that is really easy to implement (and actually I'm deleting that "primary-medium" since it doesn't make sense and change it back as how it was). But let us know if the design is revisit at some point so we can change that.
> 
> 1. Yes, it's a bit unclear if we have one or 2 features, please bring up this question on Wednesday because I'm not sure what the final decision was, and it's good that everyone hears it. Nonetheless, the token concepts worked on here should work in both cases.
> 2. I would be in favor of creating a token for the background color as "primary-color" is often reserved to the main action color. In any case I would rename it to make it clear that it's signifying the background color. Because a primary-color could be applied in many places, but a screen background color is more specific. Let me know if I can assist on naming.
> 3. Ah ok I got it, I thought in this membership feature there were 2 different colors. When working with on-dark or on-light themes then that could work, we just need another set of tokens for it. I wanted to avoid doing so for the sake of simplifying this token exercise. But I will had them

1. I'm not sure either. Was under the impression that it will be just one, at least for a while.
2. A background token would be awesome. I wanted to help out make the theming easier so I've been playing with Storybook configurations following [this ](https://storybook.js.org/blog/how-to-add-a-theme-switcher-to-storybook/) example, but without a theme specific background color, some things look like they are broken.
3. I've also wanted to ask about that as some components don't seem to have a dark theme (eg the SideSheet that is used in the HTRT screens, ProgressBanner and I'm sure that there are more, these are the ones that come to mind right now)

mteresa-jimenez commented:
New changes:

- Reverted the naming in the PolyButton component: back to "filled" and "outlined".
- Moved "coop" theme to Membership feature and change the color hierarchy values in it:
```
     --theme-primary-color: var(--color-red-300);
    --theme-secondary-color: var(--color-indigo-950);
    --theme-tertiary-color: var(--color-gray-50);
```
- Add new variable for screen background color: `--pl-screen-color-fill`

g3o314 reviewed (approved):
I think what we currently have is in a good enough state for now. We might discover new cases that we haven't covered, but we'll handle those as we build the feature.

victorjanin reviewed (approved):
Looking forward to bringing this forward! :)
